src/typing/statement.ml

```diff
@@ -5404,7 +5419,18 @@ and predicates_of_condition cx e = Ast.(Expression.(
     | true,
       (expr_loc, Member {
         Member._object;
-        property = Member.PropertyIdentifier (prop_loc, ({ Ast.Identifier.name= prop_name; comments= _ } as id));
+        property = (
+          Member.PropertyIdentifier (prop_loc, ({ Ast.Identifier.name= prop_name; comments= _ }))
+          | Member.PropertyExpression (prop_loc, Ast.Expression.Literal {
+            Ast.Literal.value = Ast.Literal.String prop_name;
+            _;
+          })
+          | Member.PropertyExpression (prop_loc, Ast.Expression.Literal {
+            Ast.Literal.value = Ast.Literal.Number _;
+            raw = prop_name;
```

Comment:
Should probably use `Dtoa.ecma_string_of_float` as in https://github.com/facebook/flow/pull/7593 PR